doc/mkdocs/docs/api/macros/json_diagnostics.md

```diff
@@ -0,0 +1,67 @@
+# JSON_DIAGNOSTICS
+
+```cpp
+#define JSON_DIAGNOSTICS /* value */
+```
+
+This macro enables [extended diagnostics for exception messages](../../home/exceptions.md#extended-diagnostic-messages).
+Possible values are `1` to enable or `0` to disable (default).
+
+When enabled, exception messages contain a [JSON Pointer](../json_pointer/json_pointer.md) to the JSON value that
+triggered the exception. Note that enabling this macro increases the size of every JSON value by one pointer and adds
+some  runtime overhead.
+
+The diagnostics messages can also be controlled with the CMake option `JSON_Diagnostics` (`OFF` by default) which sets
+`JSON_DIAGNOSTICS` accordingly.
+
+## Default definition
+
+The default value is `0` (extended diagnostics are switched off).
+
+```cpp
+#define JSON_DIAGNOSTICS 0
+```
+
+When the macro is not defined, the library will define it to its default value.
+
+## Notes
+
+!!! warning
+
+    As this macro changes the definition of the `basic_json` object, it MUST be defined in the same way globally, even
+    across different compilation units; DO NOT link together code compiled with different definitions of
+    `JSON_DIAGNOSTICS` as this is a violation of the One Definition Rule and will cause undefined behaviour.
```

Comment:
Unrelated to this PR:

I've been wondering if this should be mitigated by using inline namespaces. Do you care about ABI-stability in general? Maybe something to consider for 4.0. (Is there an established place to track 4.0 stuff? Issue, discussion, ...)

Reply:
Idea:
Add a file `Roadmap.md` to the top-level of the repo.

Use case:
- I'm working on a PR.
- I have an idea about related or unrelated additions/changes/etc. 
- I add it to `Roadmap.md` and it is preserved for the future.

Maybe have sections for definitely planned things ("Remove X in 4.0.") and more vague ideas ("Explore using inline namespaces to address ABI issues.").

Reply:
Wouldn't be issues or discussions be better ways to do so? I'm asking because "editing" a file via PRs is very cumbersome, and maybe discussions about a roadmap are then mixed with the actual PR.

Reply:
I'd say both. Having one, well organized place seems really helpful to me.

Some things may need discussion others are really just reminders to the author. It really depends on the category. If it's, for example, a note about a deprecation the discussion belongs in the PR anyway.

Reply:
Moved to #3453.